test/retry_test.go

```diff
@@ -146,8 +146,14 @@ spec:
 		if _, found := podNames[p.Name]; !found {
 			t.Errorf("BUG: TaskRunStatus.RetriesStatus did not report pod name %q", p.Name)
 		}
-		if p.Status.Phase != corev1.PodFailed {
-			t.Errorf("BUG: Pod %q is not failed: %v", p.Name, p.Status.Phase)
+		// Check each container in the pod, rather than the phase, since the phase doesn't update to a terminal state instantly.
```

Comment:
🎉 thank you for adding the explicit check!

Reply:
If the check against `p.Status.Phase` still worked (without a sleep - in my local testing, it did work fine with a `2s` delay before the `Pods(namespace).List(...)` call), I would find the each-container check to be overkill, but hey. =)